### DISCUSSION TITLE
How to cast a value of a column before it is used with the like() filter?

### DISCUSSION THREAD

**Discussion by TomasHubelbauer** [supabase/supabase#3836] [Questions] [answered]
Hi, I have a table with a `create_at` column of type `timestamptz`. I am attempting to use `like` on its value like this:

```js
await supabase.from('table').select().like('created_at', 'stuff%');
```

However, I get this error:

```
No operator matches the given name and argument types. You might need to add explicit type casts.
operator does not exist: timestamp with time zone ~~ 42883
```

I understand I can't directly use `like` on `timestamptz`, so I would like to first convert the column's value to a string and then use `like` on the result of the conversion. However, I am not finding a good sample of doing casts.

I am using supabase-js.

**Comment by steve-chavez** [comment, marked as answer]
You can use a computed column in this case. See this previous discussion https://github.com/supabase/supabase/discussions/3582#discussioncomment-1497587

**Reply by TomasHubelbauer** [reply]
I think the answer to this is "yes", but just to be completely sure, in the web UI of Supabase, this would be the alpha feature Functions under the Table category, correct?

**Reply by steve-chavez** [reply]
@TomasHubelbauer For creating computed columns as showed on https://github.com/supabase/supabase/discussions/3582#discussioncomment-1497587 you have to use the SQL Editor and run raw SQL, right now the Function Editor UI doesn't support adding parameters with table types.
